modules/extensions-soap-support/src/main/java/org/mule/runtime/module/extension/soap/internal/loader/SoapInvokeOperationDeclarer.java

```diff
@@ -108,17 +113,36 @@ void declare(ConfigurationDeclarer configDeclarer, ClassTypeLoader loader, Set<E
   }
 
   private void declareMetadata(OperationDeclarer operation, ClassTypeLoader loader) {
-    ImmutableMap.Builder<String, Supplier<? extends InputTypeResolver>> inputResolver = ImmutableMap.builder();
-    inputResolver.put(BODY_PARAM, InvokeRequestTypeResolver::new);
-    inputResolver.put(HEADERS_PARAM, InvokeInputHeadersTypeResolver::new);
-    inputResolver.put(ATTACHMENTS_PARAM, InvokeInputAttachmentsTypeResolver::new);
-    DefaultMetadataResolverFactory factory = new DefaultMetadataResolverFactory(InvokeKeysResolver::new,
-                                                                                inputResolver.build(),
-                                                                                InvokeOutputTypeResolver::new,
+    Map<String, Supplier<? extends InputTypeResolver>> inputResolver =
+        ImmutableMap.<String, Supplier<? extends InputTypeResolver>>builder()
+            .put(BODY_PARAM, InvokeRequestTypeResolver::new)
+            .put(HEADERS_PARAM, InvokeInputHeadersTypeResolver::new)
+            .put(ATTACHMENTS_PARAM, InvokeInputAttachmentsTypeResolver::new)
+            .build();
+    TypeKeysResolver keysResolver = new InvokeKeysResolver();
+    OutputTypeResolver outputResolver = new InvokeOutputTypeResolver();
+    DefaultMetadataResolverFactory factory = new DefaultMetadataResolverFactory(() -> keysResolver,
+                                                                                inputResolver,
+                                                                                () -> outputResolver,
                                                                                 NullMetadataResolver::new);
     operation.withModelProperty(new MetadataResolverFactoryModelProperty(() -> factory));
     operation.withModelProperty(new MetadataKeyIdModelProperty(loader.load(WebServiceTypeKey.class), KEYS_GROUP,
-                                                               SOAP_INVOKE_METADATA_CATEGORY));
+                                                               keysResolver.getCategoryName()));
```

Comment:
@alegmarra the value hardcoded here was added when the metadata cache was implemented but didn't match the getCategoryName() returned by the InvokeKeysResolver. Therefore I have also fixed that.